src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -34,21 +36,23 @@
      */
     public WorkerAddCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_PAY, PREFIX_ADDRESS, PREFIX_ROLE);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_PAY, PREFIX_ADDRESS, PREFIX_ROLE,
+                        PREFIX_UNAVAILABILITY);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_PAY)
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_PAY, PREFIX_ROLE)
```

Comment:
I think role is an optional argument so perhaps no need to check if it is present.
Actually since we are on this topic, I think pay isn't either? But if these are necessary then perhaps updating the User Guide will do.

Reply:
Mm I just thought that it made sense for all workers to have a role so I included it as a compulsory field, but we can take it out also I guess? So if a worker doesn't have any roles specified, then we can assign them to any role in a shift
If we're going to be implementing the Pay calculation feature then we should make Pay compulsory

Reply:
I think no role should be unable to assign to anything, kind of like when someone new join and havent do the training I guess. Makes sense for pay to be compulsory, so agreed there.

Reply:
Ah ok agreed for role! Will make the change